src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -477,6 +489,7 @@ func overprovisioned(developerMode bool, limits corev1.ResourceList) []string {
 		"--default-log-level=info",
 		"--reserve-memory 0M",
```

Comment:
I thought not having reserve-memory meant TLS didn't work? Or has memory already accounted for that?

Reply:
Yeah, they have been different. I know `reserve-memory` is set when `memory` isn't to reserve memory for the OS to use. Since we use `memory` in one case I don't specify `reserve-memory` then, but I'm not sure what happens if `memory` is set to the full memory capacity of the machine.

```
 ("reserve-memory", bpo::value&lt;std::string&gt;(),  "memory reserved to OS (if --memory not specified)")
```
https://github.com/vectorizedio/seastar/blob/master/src/core/reactor.cc#L3431-L3432

Reply:
I don't know. If non-developerMode works fine with TLS then all good. I never got to the bottom of it, but when I was testing I recall that we added --reserve-memory when rpk was taught to pass in TLS. My guess what that it consumes some memory before forking. I think we tested with 1 core and 1GiB - 1GiB being the minimum per core (2GiB is recommended), and it failed.

Reply:
As far as I remember, seastar have some hiccup in our CI where we are constraint to only 100Mb in `resource.request.memory` and DeveloperMode on. In such low constraint the seastar initialization failed.